### ISSUE TITLE
[CSL-373] Constants tuning: data for researchers

### ISSUE THREAD

**Issue by jagajaga** [input-output-hk/cardano-sl#743]
<blockquote>@georgeee</blockquote>

Provide researchers with data they need to estimate constants.

From Bernardo:

A. Can you tell us an estimate of the:
# size of a block header
# size of a transaction
# size of a block signature (i believe it’s 512 bits but would like to confirm)

B. We also need to know how long it takes  in an amazon ec2 instance to:
# generate a block header
# generate/verify signatures
# generate commitments, generate pvss shares, open/verify commitments
# run fullow-the-satoshi
# verify that a transaction is valid with respect to the current UTXO set
# verify that a block is valid

We need these data to estimate the cost of running the protocol.

Any preliminar results on estimating the runtime of these operations would be very helpful, even if they don’t reflect the full performance that can be achieved using a proper optimized implementation of the database and all the crypto primitives. We need that to estimate an upper bound on the cost of running the protocol, not to show how efficient the protocol can be, so the *not optimal values will be enough*.

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

From Bernardo:

Can you tell us an estimate of the size of a block header, the size of a transaction and the size of a block signature (i believe it’s 512 bits but would like to confirm)? We also need to know how long it takes  in an amazon ec2 instance to generate a block header, generate/verify signatures, generate commitments, generate pvss shares, open/verify commitments, run fullow-the-satoshi, verify that a transaction is valid with respect to the current UTXO set and verify that a block is valid. We need these data to estimate the cost of running the protocol.

Any preliminar results on estimating the runtime of these operations would be very helpful, even if they don’t reflect the full performance that can be achieved using a proper optimized implementation of the database and all the crypto primitives. We need that to estimate an upper bound on the cost of running the protocol, not to show how efficient the protocol can be, so the not optimal values will be enough.

**Comment by jagajaga** [comment]
<blockquote>@gromak</blockquote>

> Can you tell us an estimate of the size of a block header, the size of a transaction and the size of a block signature (i believe it’s 512 bits but would like to confirm)?
It highly depends on CSL-124 (and probably some other issues).

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

And CSL-48

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

But they told us *not optimal values will be enough*

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

George: So, let's schedule all this data be provided somewhere during next week. If we see ourselves failing to provide this data, we would inform you by Wed, 28th.

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

Regarding how much time particular action takes, we need to think about how reliably measure this values on server setup. And probably we should do it after DB layer fully established (this week) with BTC UTXO for example (cause follow the satoshi seems to be the biggest part here and it's now `O(|UTXO|)` AFAIK )

**Comment by jagajaga** [comment]
<blockquote>@georgeee</blockquote>

Postponed due to we probably won't have time to do this task
